src/fetchtastic/download/client_app.py

```
@@ -0,0 +1,1128 @@
+            try:
+                if not self._is_asset_complete_for_target(asset_path, asset):
+                    return False
+            except OSError:
+                return False
+        return True
+
+    def cleanup_old_versions(
+        self,
+        keep_limit: int,
+        cached_releases: list[Release] | None = None,
+        keep_last_beta: bool = False,
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)